starknet_py/net/client_models.py

```diff
@@ -153,7 +153,7 @@ class TransactionReceipt:
     block_number: Optional[int] = None
     version: int = 0
     actual_fee: int = 0
-    rejection_reason: Optional[str] = None
+    rejection_reason: Optional[Dict[str, Any]] = None
```

Comment:
IIR I encountered a validation error because it's effectively a `dict`.

Reply:
It is a dict in the gateway response but just a string in rpc. That's why I'd like to have rejection_reason as a string in `TransactionReceipt` (base class) since it's present in both, and then also have a `code` in `GatewayTransactionReceipt`. From my quick tests it seems like this dict only contains these two keys.